Assets/Scripts/Tanks/EnemyController.cs

```diff
@@ -0,0 +1,116 @@
+﻿using UnityEngine;
+using Tank;
+
+namespace Enemy
+{
+    public class EnemyController : TankController
+    {
+        private TankController target;
+        [SerializeField]
+        private float visionConeAngle = 45f;
+
+        [SerializeField]
+        private float shootRange = 10f;
+
+        [SerializeField]
+        private float movementDistance = 15f;
+        private Vector3 moveDir;
+        private Vector3 distanceCheckPos;
+        private EnemyStates currState;
+
+        private void Update()
+        {
+            switch (currState)
```

Comment:
See more examples of State Pattern and encapsulate all behavior for a particular state in one state class